azure-pipelines.yml

```diff
@@ -268,3 +268,17 @@ jobs:
       parameters:
         jobName: Correctness_Rebuild
         configuration: Release
+
+- job: Correctness_Analyzers
+  pool:
+    name: NetCore1ESPool-Public
+    demands: ImageOverride -equals Build.Windows.Amd64.VS2022.Pre.Open
+  timeoutInMinutes: 25
+  steps:
+    - template: eng/pipelines/checkout-task.yml
+
+    - task: PowerShell@2 
+      displayName: Build with analyzers
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -restore -build -configuration Release -prepareMachine -ci -binaryLog -runAnalyzers:$true -warnAsError:$true -properties "/p:RoslynEnforceCodeStyle=true"
```

Comment:
Do we need to build using both Debug and Release? I can imagine an analyzer reporting a diagnostic inside `#if DEBUG` that won't be caught if we're building in Release.

Reply:
I don't think its worth running both to be honest, we've already cut down our CI a fair amount to reduce chances for flakiness.  However I do see that we have wayyy more `#if DEBUG` than release ones, so I'm definitely open to running this in Debug instead.

Reply:
The compiler uses debug pragmas a lot. On the IDE side we use significantly less (about 10% of what the compiler does) but we still use them. However, all of our Debug pragmas are _additive_ so if we run the analyzer leg in Debug I think that should be all we need.